### DISCUSSION TITLE
Does there exist an example of using this as a linked external library?

### DISCUSSION THREAD

**Discussion by gclimon** [modern-fortran/neural-fortran#164] [Q&A]
Every example here I've found uses this as a module, rather than a linked library, which does not build/link correctly with the CESM (https://escomp.github.io/CESM/versions/cesm2.1/html/) model I'm currently trying to use this with. However, it is my understanding that the point of building this library on my system is that I would be able to just link to it (the lib{}.a files in /neural-fortran-main/build/lib/) and not need the explicit 'use' statements that imply it being used solely as a compiled module. Has anybody had any luck with this and could share an example or guidance? When I link the library files to my CESM build as an external library, it is identified, yet calling the internal functions does not work.

**Comment by milancurcic** [comment]
Hi @gclimon,

At compile time (i.e. when you compile CESM with neural-fortran or any other Fortran library with modules as a dependency), you need to provide the path to the compiled modules in the include path (`-I/path/to/my/modules`).

At link time, you only need to pass the compiled library files (`*.a`) to the linker.

I don't have experience with CESM but I've built other similar models many times and they're all similar but somewhat different. Can you provide your build log and the error messages that you get when you try to use neural-fortran from CESM?

**Reply by gclimon** [reply]
Hi @milancurcic, 

Thanks for the quick response. I've attempted to add the include path(s) to the /include/ directory in the neural-fortran build. I admit linking and including external modules is a bit outside of my fortran comfort zone. However, can you clarify two things: Do I need just the path to the module directory once (-I/path/to/nf-build/include) or do I need to directly specify all of the modules in /path/to/include/? Second, do I stitll use the 'use nf' statement internally? I assume yes. 

Regardless, I still have build issues. I've attached my build log with the errors as well as my Macros.make file which has been updated with the include (FFLAGS) and linking (SLIBS) flags. 

Since the issues seem to be associated with undefined refereences to aspects of json-fortran and h5fortran, I also tried with an additional include path -I(path/to/nf-build/_deps/h5fortran-build/include) but still have the same issues. Is there another include path for json-fortran that I'm missing? 

For context, the only relevant point in which my CESM SourceMods use nf at the moment is just a 'use nf' statement for testing so I don't include these files in the attachment.
[neural-fortran-cesm-logs.tar.gz](https://github.com/modern-fortran/neural-fortran/files/13392196/neural-fortran-cesm-logs.tar.gz)
